### PR TITLE
ZEN-4923: Added unirest-java dependency (version 2.3.10)

### PR DESCRIPTION
**Pull request by nbhusare**
1. Genflow pom: Added dependency under 'dependencymanagenent'
2. reprezen-openapi-gentemplates: Added dependency to the pom file.

### CODE REVIEW

**Comment by ghillairet** [comment]
@nbhusare Why do we need this library in GenFlow? If it is needed in API Studio for Confluence it should be added in this pom https://github.com/ModelSolv/RepreZen/blob/master/com.reprezen.gentemplates.apistudio/pom.xml#L16

**Comment by tedepstein** [comment]
> @nbhusare Why do we need this library in GenFlow? If it is needed in API Studio for Confluence it should be added in this pom https://github.com/ModelSolv/RepreZen/blob/master/com.reprezen.gentemplates.apistudio/pom.xml#L16

@ghillairet , How will a dependency added to that POM be processed in API Studio? Will API Studio copy this GenTemplates POM to the project /lib folder along with the JAR? Will the POM  be added as a dependency to the project POM so it will be included in code generation from API Studio and from a maven build?

**Comment by ghillairet** [comment]
@tedepstein Yes it will have to be copied in the lib folder. It will be added to the maven classpath using the shared folder property. But if it has also dependencies, all of them have to be added as well.

If we add it to GenFlow like that, then it will have to be added as well in the uber bundles pom.

Also question why choose a version from 2016?

**Comment by ghillairet** [comment]
If I look here https://github.com/Kong/unirest-java

The version is 2.3 and they have an artefact that includes all dependencies

```
<dependency>
    <groupId>com.konghq</groupId>
    <artifactId>unirest-java</artifactId>
    <version>2.3.10</version>
    <classifier>standalone</classifier>
</dependency>
```

File: modules/reprezen-openapi-gentemplates/pom.xml [hunk 1]
```diff
@@ -36,10 +36,5 @@
             <groupId>com.reprezen.genflow</groupId>
             <artifactId>swagger-ui</artifactId>
         </dependency>
-        <dependency>
-            <groupId>com.konghq</groupId>
-            <artifactId>unirest-java</artifactId>
-            <classifier>standalone</classifier>
-        </dependency>
```
Comment (tedepstein) [inline modules/reprezen-openapi-gentemplates/pom.xml]: As I understood it, `modules/reprezen-openapi-gentemplates/pom.xml ` has dependencies specific to the OpenAPI GenTemplates provided by RepreZen.  `modules/genflow-api-openapi/pom.xml ` has dependencies that apply to _all_ OpenAPI GenTemplates from _all_ providers. Is that right? If so, wouldn't it work to add these dependencies to `modules/reprezen-openapi-gentemplates/pom.xml ` , so they are specific to RepreZen GenTemplates?

**Comment by ghillairet** [comment]
Also see https://github.com/RepreZen/APIDocsForConfluence/pull/34

File: modules/reprezen-openapi-gentemplates/pom.xml (hunk 1 above)
Comment (ghillairet) [reply modules/reprezen-openapi-gentemplates/pom.xml]: `reprezen-openapi-gentemplates` is the group that contains all our gentemplates. `genflow-api-openapi` contains the actual classes to create OpenAPI gentemplates.

**Pull request merged**